pkg/arvo/sys/hoon.hoon

```diff
@@ -1286,10 +1286,10 @@
       ?.  (mor n.a n.b)
         $(a b, b a)
       ?:  =(n.b n.a)
-        [n.a $(a l.a, b l.b) $(a r.a, b r.b)]
+        a(l $(a l.a, b l.b), r $(a r.a, b r.b))
       ?:  (gor n.b n.a)
-        %-  uni(a $(a l.a, b [n.b l.b ~]))  $(b r.b)
-      %-  uni(a $(a r.a, b [n.b ~ r.b]))  $(b l.b)
+        %-  uni(a $(a l.a, b b(r ~)))  $(b r.b)
```

Comment:
`$(a l.a, b b(r ~))` can be further reduced to `$(a l.a, r.b ~)`.  That would take us all the way to making the smallest possible incision into these nouns, so I think it would be better -- maybe a bit clearer too.

Reply:
Ah yes, some of those `b b(...` were bothering me so that's definitively better.  We can push it a bit further and reduce this `$(a a(l $(a l.a, r.b ~)), b r.b)` to this `$(l.a $(a l.a, r.b ~), b r.b)`